src/menu.scss

```diff
@@ -5,352 +5,331 @@
-    -webkit-box-orient: vertical;
+    list-style: none;
+    box-shadow: $fd-menu-box-shadow;
+    border-radius: $fd-menu-border-radius;
+
+    &--no-shadow {
+      box-shadow: none;
+    }
   }
 
   &__sublist {
     @include fd-reset();
 
-    left: 100%;
-    right: initial;
     position: absolute;
     top: 0;
+    left: 100%;
     z-index: 2;
-    margin: 0.25rem 0 0 -0.25rem; // offset of submenu
+    margin: $fd-menu-sublist-offset;
+    width: 100%; // to fix IE issue
 
     @include fd-rtl() {
       right: 100%;
       left: initial;
       margin: 0.25rem -0.25rem 0 0;
     }
 
-    width: 100%;  // to fix IE issue
-
     &[aria-hidden='true'] {
       display: none;
     }
   }
 
   &__list > li {
     position: relative; // submenu's positioning relative to parent item
-    width: 100%;  // to ensure in mobile mode the width of parent is taken fully
+    width: 100%; // to ensure in mobile mode the width of parent is taken fully
   }
 
   &--compact {
-    @include fd-reset();
-
     .#{$block}__link {
-      // for desktop mode
       padding: 0.375rem 0.75rem 0.375rem 0.5rem; // line-height=20px so 2rem or 32px-20px=12px/2= 6px or 0.375rem each
-      height: 2rem;
-    }
-
-    .#{$block}__addon-before {
-      min-width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
-      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
-      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
-      margin: 0 0 0 -0.5rem;
-    }
+      height: $fd-menu-link-height-compact;
 
-    .#{$block}__addon-after {
-      min-width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
-      width: 2rem; // to account for item right padding of 12px or 0.75rem so 2rem-0.75rem = 1.25rem
-      height: 1.25rem; // to account for item top padding of 6px or 0.375rem so 2rem-0.375rem-0.375rem = 1.25rem
-      margin: 0 -0.75rem 0 0;
-    }
+      .#{$block}__addon-before,
+      .#{$block}__addon-after {
+        min-width: $fd-menu-icon-width-compact;
+        width: $fd-menu-icon-width-compact;
+      }
 
-    // handle icons rtl modes
-    @include fd-rtl() {
       .#{$block}__addon-before {
-        margin: 0 -0.75rem 0 0; // since before and after icons switch places, their margins also switch
+        margin: 0 0 0 -0.5rem;
       }
 
       .#{$block}__addon-after {
-        margin: 0 0 0 -0.5rem; // since before and after icons switch places, their margins also switch
+        margin: 0 -0.75rem 0 0;
+      }
+
+      @include fd-rtl() {
+        .#{$block}__addon-before {
+          margin: 0 -0.75rem 0 0; // since before and after icons switch places, their margins also switch
+        }
+
+        .#{$block}__addon-after {
+          margin: 0 0 0 -0.5rem; // since before and after icons switch places, their margins also switch
+        }
```

Comment:
These should be vars, especially because they are reused in RTL and if the values change in future it will be easy to forget to update the values in the RTL block